GH#27862 Put must gather audit logs back

mburke5678 opened the pull request:
https://github.com/openshift/openshift-docs/issues/27862

Audit logs are no longer collected by default to reduce the size of the log bundle, this was suppressed via [this commit](https://github.com/openshift/must-gather/commit/7b3fff50ac06fa3dd288aaf55679d46a53db8a84#diff-bad6eb3f5953b7f47f014fd8b04a48470c1d64d94bbd29566af842f0b60c909c).

If needed, the audit logs can be gathered again on the following manner:

$ oc adm must-gather -- /usr/bin/gather_audit_logs

Previews: 
New text and organization in [About the must-gather tool](https://deploy-preview-38415--osdocs.netlify.app/openshift-enterprise/latest/support/gathering-cluster-data.html#about-must-gather_gathering-cluster-data)
New section on [Gathering audit logs](https://deploy-preview-38415--osdocs.netlify.app/openshift-enterprise/latest/support/gathering-cluster-data.html#gathering-data-audit-logs_gathering-cluster-data). 
Same section in the [Viewing Audit Logs assembly](https://deploy-preview-38415--osdocs.netlify.app/openshift-enterprise/latest/security/audit-log-view.html#gathering-data-audit-logs_audit-log-view) with `ifevals` comments to change the intro paragraph.

mburke5678 commented:
@pamoedom PTAL

pamoedom commented:
Hi @mburke5678 , thanks for the changes, appreciated, I have a pair of comments/suggestions:

First of all, in the `About the must-gather tool` section, it still mentions `Audit logs` as part of default gathering info, how about adding a note at the bottom that redirects to specific `Gathering audit logs` section explaining that is not included by default to alleviate the size of the logs.

Secondly, within the `Gathering audit logs` itself, the first command example says the following:
~~~
For example, the following command gathers both the default cluster data and information specific to KubeVirt:
$ oc adm must-gather -- /usr/bin/gather_audit_logs
~~~
but that's incorrect, that command will only extract the audit logs and not the rest of default cluster data, in order to achieve that we need to include also the following command:
~~~
$ oc adm must-gather -- /usr/bin/bash -c '/usr/bin/gather ; /usr/bin/gather_audit_logs'
~~~
Maybe we should include here also a note about how extracting them together if needed, what do you think?

Best Regards.

mburke5678 commented:
@pamoedom Great comments. I thought about removing audit logs from the list in the About section but feared burying the information. I can see your concern though. I can make that change. 
As for the command in the Gathering section, that was a copy and paste error. Sorry!

pamoedom commented:
> @pamoedom Great comments. I thought about removing audit logs from the list in the About section but feared burying the information. I can see your concern though. I can make that change. As for the command in the Gathering section, that was a copy and paste error. Sorry!

Perfect, thanks @mburke5678.

mburke5678 commented:
@masayag @ yossisegev 
Hello. You [reviewed the documentation for the `oc adm must-gather` tool](https://github.com/openshift/openshift-docs/pull/16567) when it was first added to the docs. Would you be appropriate to review this change? Apparently, the [audit logs were removed](https://github.com/openshift/must-gather/commit/7b3fff50ac06fa3dd288aaf55679d46a53db8a84#diff-bad6eb3f5953b7f47f014fd8b04a48470c1d64d94bbd29566af842f0b60c909c) from the default set of information gathered by must-gather. We got a request to document how to gather them.

pamoedom commented on modules/about-must-gather.adoc:
```diff
@@ -13,11 +13,12 @@
 
 You can specify one or more images when you run the command by including the `--image` argument. When you specify an image, the tool collects data related to that feature or product.
 
+You can alternatively collect audit logs using the `-- /usr/bin/gather_audit_logs` flag. Audit logs are not collected as part of the default set of information to reduce the size of the files. 
+
```
Hi @mburke5678 , I think it would be better to move this paragraph to a bottom NOTE section instead, how about something like this?
~~~
[NOTE]
====
Audit logs are no longer gathered by default to reduce the folder size, for more info please refer to section xref:../support/gathering-cluster-data.html#gathering-data-audit-logs_gathering-cluster-data[Gathering audit logs].
====
~~~
NOTE: please double check the xref path, not sure if I have used it correctly.

mburke5678 replied:
@pamoedom Good idea to use a link. However, we cannot add links to modules (for reasons that are not clear to me). I added a _as described in the following sections_. And reorganized a lit to tighten up the sentences, particularly the sentence on the --image argument.

mburke5678 commented:
@yossisegev You [reviewed the Must Gather docs](https://github.com/openshift/openshift-docs/pull/16567#issuecomment-532150006) when they were added. Is this PR that you can review? How to use `oc adm must-gather` to collect audit logs.

yossisegev commented:
@dbasunag@redhat.com @ibesso@redhat.com
Looks like this is your domain.

pamoedom commented:
/lgtm

ibesso-rh commented:
@mburke5678 , here is my two cents:
* maybe we should add to the docs what the command help specifies, i.e. if no image is used the default image:
_\# Gather information using the default plug-in image and command, writing into ./must-gather.local.<rand>
  oc adm must-gather_
* for the command example for audit logs, why not add the full command, not just the additional suffix?
* do we want to add an indication about the pod's namespace/name?

mburke5678 commented:
@ibesso-rh 
How does the naming work? It appears to create a randomly-named pod in a randomly-named namespace. 

```
openshift-must-gather-vtvjl                        must-gather-qfq45                                             2/2     Running     0             45s
openshift-must-gather-vtvjl                        must-gather-zg4qw                                             2/2     Running     0             45s

```

pamoedom commented:
/lgtm

mburke5678 commented:
@ibesso-rh PTAL. I think I have addressed all of your concerns in [my latest commit](https://github.com/openshift/openshift-docs/pull/38415/commits/6b002e7497dbae54e875d34d0bac7ee6558e3b53).

mburke5678 commented:
@ibesso-rh PTAL

ibesso-rh commented:
/lgtm

ibesso-rh commented:
> @ibesso-rh How does the naming work? It appears to create a randomly-named pod in a randomly-named namespace.
> 
> ```
> openshift-must-gather-vtvjl                        must-gather-qfq45                                             2/2     Running     0             45s
> openshift-must-gather-vtvjl                        must-gather-zg4qw                                             2/2     Running     0             45s
> ```

@mburke5678 , I believe it is random.
What I meant in my suggestion is just to mention that, so the user/admin will know the process in the background, i.e. pods created and eventually destroyed.

ahardin-rh commented on modules/gathering-data-audit-logs.adoc:
```diff
@@ -0,0 +1,55 @@
+
+endif::support[]
+ifdef::viewing[]
+You can use the xref:../support/gathering-cluster-data.html#about-must-gather_gathering-cluster-data[must-gather tool] to collect the audit logs for debugging your cluster, which you can review or send to Red Hat Support.
```
s/.html/.adoc

ahardin-rh commented on modules/gathering-data-audit-logs.adoc:
```diff
@@ -0,0 +1,55 @@
+----
+
+. Create a compressed file from the `must-gather` directory that was just created in your working directory. For example, on a computer that uses a Linux
+operating system, run the following command:
```
looks like a strange line break here after the word "Linux"

ahardin-rh commented on modules/gathering-data-audit-logs.adoc:
```diff
@@ -0,0 +1,55 @@
+----
+$ tar cvaf must-gather.tar.gz must-gather.local.472290403699006248 <1>
+----
+<1> Make sure to replace `must-gather-local.472290403699006248` with the actual directory name.
```
s/Make sure to replace/Replace

mburke5678 commented:
/cherrypick enterprise-4.6

mburke5678 commented:
/cherrypick enterprise-4.7

mburke5678 commented:
/cherrypick enterprise-4.8

mburke5678 commented:
/cherrypick enterprise-4.9

mburke5678 commented:
/cherrypick enterprise-4.10
